lib/stats/skills.rb

```diff
@@ -0,0 +1,78 @@
+require "ostruct"
+
+module Skills
+  # extended function, now takes INT, Symbol, String but not shorthand Symbol, String
+  # Skills.to_bonus(Skills.combatmaneuvers), Skills.to_bonus(5),
+  # Skills.to_bonus(:combat_maneuvers), Skills.to_bonus('combat_maneuvers') as examples
+  def self.to_bonus(ranks)
+    case ranks
+    when Integer
+      bonus = 0
+      while ranks > 0
+        if ranks > 40
+          bonus += (ranks - 40)
+          ranks = 40
+        elsif ranks > 30
+          bonus += (ranks - 30) * 2
+          ranks = 30
+        elsif ranks > 20
+          bonus += (ranks - 20) * 3
+          ranks = 20
+        elsif ranks > 10
+          bonus += (ranks - 10) * 4
+          ranks = 10
+        else
+          bonus += (ranks * 5)
+          ranks = 0
+        end
+      end
+      bonus
+    when String, Symbol
+      Infomon.get("skill.%s.bonus" % ranks)
+    else
+      echo "You're trying to move the cheese!"
+    end
+  end
+
+  @@skills = %i(two_weapon_combat armor_use shield_use combat_maneuvers edged_weapons blunt_weapons two_handed_weapons ranged_weapons thrown_weapons polearm_weapons brawling ambush multi_opponent_combat physical_fitness dodging arcane_symbols magic_item_use spell_aiming harness_power elemental_mana_control mental_mana_control spirit_mana_control elemental_lore_air elemental_lore_earth elemental_lore_fire elemental_lore_water spiritual_lore_blessings spiritual_lore_religion spiritual_lore_summoning sorcerous_lore_demonology sorcerous_lore_necromancy mental_lore_divination mental_lore_manipulation mental_lore_telepathy mental_lore_transference mental_lore_transformation survival disarming_traps picking_locks stalking_and_hiding perception climbing swimming first_aid trading pickpocketing)
+  # todo: lich up through 5.6.2 returns rank as integer - should we extend to include bonus?
```

Comment:
we should do `<skill>_bonus` imo as this is more "Rubyist"
`trading_bonus`
`two_weapon_combat_bonus`